### ISSUE TITLE
Add support for user defined operators solc 0.8.19

### ISSUE THREAD

**Issue by 0xalpharush** [crytic/slither#1681]
### Describe the desired feature

https://blog.soliditylang.org/2023/02/22/solidity-0.8.19-release-announcement/

**Comment by montyly** [comment]
See `AST Changes` in https://blog.soliditylang.org/2023/02/22/user-defined-operators/

We can take different options on how we will handle the user defined operators, but I think we should replace the `+` / `*` with a call to the underlying function in the IR

**Comment by montyly** [comment]
wip in https://github.com/crytic/slither/pull/1684

**Comment by PaulRBerg** [comment]
I have tried running Slither in a project which uses user-defined operators (the ones defined in [PRBMath](https://github.com/PaulRBerg/prb-math)), and I got this error:

```text
full_name_split = f["function"]["name"].split(".")
KeyError: 'function'
ERROR:root:Error in
```

Here is the full log: https://app.warp.dev/block/SQKsqC1VgU7v4bzfKNlDg9

Is this error due to operators, or is it unrelated?

**Comment by 0xalpharush** [comment]
This is due to user defined operators. We had started support in https://github.com/crytic/slither/pull/1684 but it proved difficult and is not yet ready

**Comment by PaulRBerg** [comment]
If this issue is closed, will operators be supported in v0.9.7?

**Comment by smonicas** [comment]
The operators are supported so if you have any issues that you think are related to user defined operators open a new issue. For prb-math there are other issues unrelated to operators basically before it was failing on operators after that was implemented new issues arose (#1989).
